### PR TITLE
DAOS-9957 test: Test set_attr, list_attr, get_attr with async mode in…

### PR DESCRIPTION
**Pull request by shimizukko**
… pool/attribute.py

Test set_attr, list_attr, get_attr with async mode in pool/attribute.py
The original code was only testing set_attr.
Also use CallbackHandler from our infrastructure code.

Skip-unit-tests: true
Test-tag: pool_attribute
Signed-off-by: Makito Kano <makito.kano@intel.com>

### CODE REVIEW

**Review by dinghwah — DISMISSED** [review]
LGTM

**Review by daltonbohning — DISMISSED** [review]
LGTM, but suggested comment

**Review by daltonbohning — APPROVED** [review]
LGTM

**Review by dinghwah — APPROVED** [review]
LGTM

File: src/tests/ftest/pool/api_attribute.py [hunk 1]
```diff
@@ -107,7 +107,10 @@ def test_pool_large_attributes(self):
         Test description: Test large randomly created pool attribute.
 
         :avocado: tags=all,daily_regression
-        :avocado: tags=regression,pool,pool_attr,attribute,large_poolattribute
+        :avocado: tags=vm
+        :avocado: tags=pool
+        :avocado: tags=pool_attribute
+        :avocado: tags=regression,pool_attr,large_pool_attribute
```
[the remark this answers is not in the document]
Comment (shimizukko) [reply src/tests/ftest/pool/api_attribute.py]: I didn't realize pool_attribute and pool_attr are duplicate. Updated the tags. Thanks!

**Comment by shimizukko** [comment]
Updated `DaosPool.get_attr()` in pydaos API, but api_attribute.py is the only test that's using this method.

**Comment by phender** [comment]
@shimizukko due to the `daos_api.py` changes additional tests need to be run.  Please make sure to include the following tests:
- container/attribute.py
- datamover/dst_create.py
- datamover/posix_prepare_props.py
- pool/attribute.py

**Comment by shimizukko** [comment]
I ran the four tests and all passed.

**Pull request merged**